packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2817,6 +2817,12 @@ export class ContainerRuntime
 				forcedFullTree,
 			} as const;
 
+			try {
+				this.summarizerNode.validateSummary();
+			} catch (error) {
+				return { stage: "generate", ...generateSummaryData, error };
+			}
+
 			continueResult = checkContinue();
 			if (!continueResult.continue) {
 				return { stage: "generate", ...generateSummaryData, error: continueResult.error };
```

Comment:
Somewhat off-topic: I feel like there should be some indication that this is a `continueResult` error.  Maybe can put it in the same spot I mentioned above on `IGeneratedSummaryStats`, like `failureReason: "invalidSummaryGenerated" | "cannotContinue"` or something

Reply:
Let's put that in the bucket of summarization improvements. I totally agree that the failures don't have enough information.